### DISCUSSION TITLE
GluonTS make_evaluation_prediction function runs endlessly on DeepAR model trained.

### DISCUSSION THREAD

**Discussion by sayedathar11** [awslabs/gluonts#1789] [Q&A] [answered]
Hiii ,
We are solving forecasting problem where target is in the range (0,1) , we have close to 89k time series in test set . While using make_evaluation_prediction function it appears that its progress stop somewhere in 3 % to 10 %  range and it runs endlessly without giving an error.

Note: We are imputing features but not target , we are letting GluonTS and make_evaluation_prediction function handle the missing dates target which are replaced by NaN 

I also tried using make_evaluation_prediction function on 33k time series  which were both in train and test  , but same problem persisted .

I am attaching screenshot of how the make_evaluation_prediction functions seems to run endlessly .
@mbohlkeschneider @lostella  can you please help .

Thanks
![deepar_infinite_loop](https://user-images.githubusercontent.com/17905011/142979583-6d0541d5-2836-4483-a46b-97b45df1ec31.PNG)
 .

**Comment by pratikgehlott** [comment]
hi, @sayedathar11, I am also facing the same issue, can anyone help me @mbohlkeschneider, @lostella

**Comment by hardik-raja-dsw** [comment]
hi, @sayedathar11, I am also facing the same issue, Appreciate if anyone can help us on same.

**Comment by mbohlkeschneider** [comment]
Hi @sayedathar11, 

thank you for raising the issue. I tried reproducing the issue with synthetic data with the snipped below, but it runs fine. 

May I ask you to modify the snipped so it produces the error? 

```
from gluonts.dataset.common import ListDataset
from gluonts.mx.trainer import Trainer
from gluonts.model.deepar import DeepAREstimator
from gluonts.evaluation.backtest import make_evaluation_predictions
import numpy as np
from tqdm.autonotebook import tqdm 

data = []
for i in range(0, 100000):
    target = np.random.random(1000)
    start = '2012-01-01'
    data.append({'start': start, 'target': target})
dataset = ListDataset(data, freq='H')

estimator = DeepAREstimator(freq='H', prediction_length=10, trainer=Trainer(epochs=1, num_batches_per_epoch=1))
predictor = estimator.train(dataset)

forecast_it, ts_it = make_evaluation_predictions(dataset, predictor)

forecasts = list(tqdm(forecast_it, total=len(dataset)))
```

**Reply by sayedathar11** [reply]
@mbohlkeschneider , I solved this issue , this issue was arrising due to present of inf (infinity) values in the data . GluonTS make_evaluation_predictions can handle NaN in targets , but I guess when there is inf values  , it just gets stuck endlessly .

**Comment by mbohlkeschneider** [comment, marked as answer]
I can confirm that `inf` values cause the issue. If you experience this issue, please check whether there are `inf` values in your data. I will think about a check/warning to catch this.  

```
from gluonts.dataset.common import ListDataset
from gluonts.mx.trainer import Trainer
from gluonts.model.deepar import DeepAREstimator
from gluonts.evaluation.backtest import make_evaluation_predictions
import numpy as np
from tqdm.autonotebook import tqdm 

data = []
for i in range(0, 1000):
    target = np.random.random(1000)
    target[-1] = np.inf
    target[-20] = np.inf
    start = '2012-01-01'
    data.append({'start': start, 'target': target})
dataset = ListDataset(data, freq='H')

estimator = DeepAREstimator(freq='H', prediction_length=10, trainer=Trainer(epochs=1, num_batches_per_epoch=1))
predictor = estimator.train(dataset)

forecast_it, ts_it = make_evaluation_predictions(dataset, predictor)

forecasts = list(tqdm(forecast_it, total=len(dataset)))
```
